Slack migration settings (#9043)
fishtown-analytics/docs.getdbt.com: website/docs/docs/deploy/job-notifications.md

That's it! Your Slack channel is now set up to receive dbt job notifications at the account level. This integration is now available throughout the account for all licensed users.

### Migrating legacy Slack notification settings

If you previously configured Slack notifications with the legacy integration, you may see a migration banner on the **Slack notifications** page.

<Lightbox src="/img/docs/deploy/migrating-legacy-slack-notifications.png" width="100%" title="Example of the Slack migration banner"/>

The banner appears when all of the following are true:

- You have existing legacy (v1) Slack notification settings.
- Your account is connected to the new Slack app (v2 OAuth is linked).
- You do not have any v2 Slack notification settings yet.

To migrate your settings, click **Migrate settings**. dbt copies your legacy Slack notification settings to the new Slack app.

You can click **Dismiss** to hide the banner for your current session. The banner appears again when you reload the page unless migration has completed successfully.

> After migration, if needed, dbt shows an informational message listing private Slack channels that still need setup. Invite the dbt bot to each listed private channel so notifications can be delivered.

Author: ```suggestion
After migration, if needed, dbt shows an informational message listing private Slack channels that still need setup. Invite the dbt Platform app to each listed private channel so notifications can be delivered.
```

That's it! Your Slack channel is now set up to receive dbt job notifications at the account level. This integration is now available throughout the account for all licensed users.

### Migrating legacy Slack notification settings

> If you previously configured Slack notifications with the legacy integration, you may see a migration banner on the **Slack notifications** page.

Reviewer: the [setup section](https://docs-getdbt-com-git-nfiann-slacknotifications-dbt-labs.vercel.app/docs/deploy/job-notifications?version=1.12#set-up-the-slack-integration-1) earlier says you must unlink the old app before linking the new one. But the migration section assumes both existed at some point. 

do we want users unlink the old integration after migrating? Is it automatic? jsut trying to understand what you should do otherwise it might cause confusion?

<Lightbox src="/img/docs/deploy/migrating-legacy-slack-notifications.png" width="100%" title="Example of the Slack migration banner"/>

The banner appears when all of the following are true:

> - You have existing legacy (v1) Slack notification settings.

Reviewer: i think v1 and v2 is more internal jargon than external - a user might not know what it means specifically. i think using words like legacy/user-linked  for v1 and account-level or dbt platform app (which we use later one) for v2 might be clearer. we should be consistent

> - Your account is connected to the <Constant name="dbt_platform" /> app (v2 OAuth is linked).

Reviewer: is v2 oauth clear for a user? is the oauth the dbt platform app?

- You do not have any v2 Slack notification settings yet.

> To migrate your settings, click **Migrate settings**. dbt copies your legacy Slack notification settings to the <Constant name="dbt_platform" /> app.

Reviewer: ```suggestion
To migrate your settings, click **Migrate settings** button in the banner. dbt copies your legacy Slack notification settings to the updated <Constant name="dbt_platform" /> app.
```

> To migrate your settings, click **Migrate settings**. dbt copies your legacy Slack notification settings to the <Constant name="dbt_platform" /> app.

Reviewer: maybe we can turn this into bullets or quick steps. eg.

To migrate your settings:
- Click **Migrate settings** button in the banner. dbt copies your legacy Slack notification settings to the updated <Constant name="dbt_platform" /> app.
- To hide the banner, click **Dismiss** to hide the banner for your current session. The banner appears again when you reload the page unless migration has completed successfully.

etc....

You can click **Dismiss** to hide the banner for your current session. The banner appears again when you reload the page unless migration has completed successfully.

> After migration, if needed, dbt shows an informational message listing private Slack channels that still need setup. Invite the <Constant name="dbt_platform" /> app to each listed private channel so notifications can be delivered.

Reviewer: this is a great section and totally get why it's mentioned here. 

i also think the private channels limitation is something people should also know earlier on too so burying it at the end means they'll migrate, wonder why notifications aren't working, and then have to troubleshoot. maybe we can add a couple of sentences explaining:

- what gets migrated — explicitly list what carries over (channels, job selections, environment filters, run status preferences — whatever it actually includes)
- what _ doesn't get carried over / what you need to do manually: i think this is where the private channels callout can go. e.g. "if any of your notification channels are private, you'll need to invite the dbt platform app to those channels after migrating. notifications aren't delivered to private channels until the dbt platform app is added. (or soemthing like that)

> <Lightbox src="/img/docs/deploy/dbt-platform-slack-invite.png" width="100%" title="Example of private channel invite guidance for the dbt Platform app"/>

Reviewer: per branding, we don't use a capital p for dbt platform
```suggestion
<Lightbox src="/img/docs/deploy/dbt-platform-slack-invite.png" width="100%" title="Example of private channel invite guidance for the dbt platform app"/>
```

That's it! Your Slack channel is now set up to receive dbt job notifications at the account level. This integration is now available throughout the account for all licensed users.

### Migrating legacy Slack notification settings

> If you previously configured Slack notifications with the legacy integration, you may see a migration banner on the **Slack notifications** page.

Reviewer: 🔥

<Lightbox src="/img/docs/deploy/migrating-legacy-slack-notifications.png" width="100%" title="Example of the Slack migration banner"/>

> Before you can migrate settings, you must unlink the legacy Slack integration and link the <Constant name="dbt_platform" /> app. Unlinking the legacy integration is a manual step, and only one Slack app can be linked at a time.

Reviewer: sugg moving line 219 up here. e.g.g
```suggestion
The banner appears when all of the following are true:

- You have notification settings from a previous Slack integration.
- Your account is connected to the <Constant name="dbt_platform" /> app.
- You have not yet configured Slack notifications with the <Constant name="dbt_platform" /> app.

Before migrating, you must unlink the legacy Slack integration and link the <Constant name="dbt_platform" /> app. Unlinking the legacy integration is a manual step, and only one Slack app can be linked at a time.

```

> The banner appears when all of the following are true:

Reviewer: if you accept above sugg, we'd prob need to remve this?

- You have notification settings from a previous Slack integration.
- Your account is connected to the <Constant name="dbt_platform" /> app.
- You have not configured Slack notification settings with the <Constant name="dbt_platform" /> app yet.

> The <Constant name="dbt_platform" /> Slack app sends job notifications to _public_ channels in your workspace. Private channels are different: notifications are not delivered there until you invite the <Constant name="dbt_platform" /> app to each private channel you use.

Reviewer: 🔥

> Migration copies your existing Slack notification settings to the <Constant name="dbt_platform" /> app, including:

Reviewer: love this section, maybe we can condense this all to this:
```suggestion

1. Click **Migrate settings** to copy your existing settings to the <Constant name="dbt_platform" /> app, including:

  - Your selected **Notification channel** and **Environment**
  - Your selected jobs
  - Your notification toggles (for example, **Succeeds**, **Warns**, **Fails**, and **Is canceled**)

2. Click **Dismiss** to hide the banner for your current session &mdash; it reappears on reload until migration completes.
3 After migration, if needed, dbt shows an informational message listing private Slack channels that still need setup.
  - If any of your channels are private, invite the <Constant name="dbt_platform" /> app to each one after migrating so notifications can be delivered. 
  
When migration succeeds, dbt hides the banner and refreshes your Slack notification settings. If migration fails, the banner remains so you can try again.
```
